File: 948B-primalSport/948B-primalSport.py
from sys import stdin
maxn = 1000006
prime = [0] * maxn
def getPrimes():
    #Encontrar el mayor factor primo de cada número
    n = 2
    while(n < maxn):
        num = n*2
        while(num < maxn):
            prime[num] = n
            #print("num",num,"= n",n)
            num += n
        num = n + 1
        #print(num,n)
        while(  num < maxn and prime[num]!= 0 ):
            num += 1
        n = num


def main():
    getPrimes()
    n = int(stdin.readline().strip())
    a = 0
    b = prime[n]
    x0 = 9999999
    print(b)
    for i in range(n-b+1,n+1):
        
        a = prime[i]
        x0 = min(i-a+1,x0)
        print("i",i,"prime[i]",prime[i],"x0",x0)
    print(x0)

main()

File: 948B-primalSport/948B-primalSport.cpp
#include <bits/stdc++.h>
#define ll long long


using namespace std;

const int maxn = 1000006;
int prime[maxn];
void getprime()
{
	//Mayor factor primo de cada numero
    int n = 2;
    while (n < maxn)
    {
        int num = n * 2;
        while (num < maxn)
        {
            prime[num] = n;
            num += n;
        }
        num = n + 1;
        while (prime[num] != 0 && num < maxn)
        {
            num++;
        }
        n = num;
    }
}
int main()
{
	//freopen("in.txt","r",stdin);
    //freopen("out.txt","w",stdout);
	getprime();
    int x0, x1, x2;
    
    while (~scanf("%d", &x2))
    {
    	//printf("%d %d\n",prime[x2],x2);
        int a = 0, b = 0;
        b = prime[x2];
        x0 = 99999999;
        for (int i = x2 - b + 1; i <= x2; i++)
        {
            a = prime[i];
            x0 = min(i - a + 1, x0);
        }
        printf("%d\n", x0);
    }
    return 0;
}
